orttraining/orttraining/core/session/training_session.cc

```diff
@@ -1000,6 +1004,9 @@ std::unordered_set<std::string> TrainingSession::GetStateTensorNames() const {
   std::unordered_set<std::string> checkpointed_tensor_names{};
   checkpointed_tensor_names.insert(
       weights_to_train_.begin(), weights_to_train_.end());
+  for (auto& p : updated_weight_names_map_) {
```

Comment:
Is there a way to update the `weights_to_train_` with the new weight names, rather than keeping around both structures?

Reply:
We are working on restructuring how we maintain the mapping between the original weight names passed and the modifications ORT does( fp16 weight name, associated optimizer state names, modified fp32 weight name if any) so that we can pass enough information to reconstruct the state in checkpointing without relying on string matching. This will be changed in that task.